share/share.go

```diff
@@ -17,6 +18,20 @@ type Share struct {
 	data []byte
 }
 
+func (s Share) MarshalJSON() ([]byte, error) {
+	return json.Marshal(s.data)
+}
+
+func (s *Share) UnmarshalJSON(data []byte) error {
+	var buf []byte
+
+	if err := json.Unmarshal(data, &buf); err != nil {
+		return err
+	}
+	s.data = buf
+	return validateSize(s.data)
```

Comment:
[blocking]

Check version

Reply:
It depends what this gets used for. Erasure coded shares can have any random bytes of 512 in length. If we implemented this check here then if we were to unmarshal an EDS encoded in json it would always return an error

Reply:
Ok, good point. And there is no way to detect if that's a parity share by only looking at the bytes. Resolving